File: src/main/java/ru/citeck/idea/index/IndexesService.java
package ru.citeck.idea.index;

import com.intellij.openapi.project.Project;
import com.intellij.psi.search.GlobalSearchScope;
import com.intellij.util.indexing.FileBasedIndex;
import lombok.RequiredArgsConstructor;

import java.util.Collection;
import java.util.List;
import java.util.stream.Collectors;
import java.util.stream.Stream;

@RequiredArgsConstructor
public class IndexesService {

    public static IndexesService getInstance(Project project) {
        return project.getService(IndexesService.class);
    }

    private final Project project;

    public Stream<IndexValue> stream(IndexKey key) {
        return FileBasedIndex
            .getInstance()
            .getValues(CiteckIndex.NAME, key, GlobalSearchScope.allScope(project))
            .stream()
            .flatMap(Collection::stream);
    }

    public List<IndexValue> list(IndexKey key) {
        return stream(key).collect(Collectors.toList());
    }
}

File: src/main/kotlin/ru/citeck/idea/search/index/IndexesService.kt
package ru.citeck.idea.search.index

import com.intellij.openapi.components.Service
import com.intellij.openapi.project.Project
import com.intellij.psi.search.GlobalSearchScope
import com.intellij.util.indexing.FileBasedIndex
import java.util.stream.Collectors
import java.util.stream.Stream

@Service(Service.Level.PROJECT)
class IndexesService(
    private val project: Project
) {

    companion object {
        @JvmStatic
        fun getInstance(project: Project): IndexesService {
            return project.getService(IndexesService::class.java)
        }
    }

    fun stream(key: IndexKey): Stream<IndexValue> {
        return FileBasedIndex
            .getInstance()
            .getValues(CiteckIndexConstants.INDEX_NAME, key, GlobalSearchScope.allScope(project))
            .stream()
            .flatMap { obj: List<IndexValue> -> obj.stream() }
    }

    fun list(key: IndexKey): List<IndexValue> {
        return stream(key).collect(Collectors.toList())
    }
}
